File: src/main/java/ninja/hudy/infosmog/repository/CountyRepository.java
package ninja.hudy.infosmog.repository;

import ninja.hudy.infosmog.model.CountyRegion;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;
import java.util.Set;

@Repository
public interface CountyRepository extends JpaRepository<CountyRegion, Long> {
    Set<CountyRegion> findByName(String name);

    Optional<CountyRegion> findByCode(Integer code);
}

File: src/main/kotlin/pl/org/smok/infosmog/repository/CountyRepository.kt
package pl.org.smok.infosmog.repository

import pl.org.smok.infosmog.model.CountyRegion
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

interface CountyRepository : JpaRepository<CountyRegion, Long> {
    fun findByName(name: String): Set<CountyRegion>

    fun findByCode(code: Int): CountyRegion
}
